packages/modules/psa/api.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import base64
+import logging
+from typing import Union
+
+from modules.common import req
+
+log = logging.getLogger("soc."+__name__)
+
+# these two lines enable debugging at httplib level (requests->urllib3->http.client)
+# You will see the REQUEST, including HEADERS and DATA, and RESPONSE with HEADERS but without DATA.
+# The only thing missing will be the response.body which is not logged.
+
+
+def requestPSAData(user_id: str,
+                   password: str,
+                   client_id: str,
+                   client_secret: str,
+                   manufacturer: str,
+                   soccalc: bool,
+                   vehicle_vin: str,
+                   vehicle: int) -> dict:
+
+    # Step1: Authentication: get access_token
+    scope = 'openid profile'
+    if (manufacturer == "Peugeot"):
+        brand = 'peugeot.com'
+        realm = 'clientsB2CPeugeot'
+        if (not client_id) or (not client_secret):
+            client_id = "1eebc2d5-5df3-459b-a624-20abfcf82530"
+            client_secret = "T5tP7iS0cO8sC0lA2iE2aR7gK6uE5rF3lJ8pC3nO1pR7tL8vU1"
+    elif (manufacturer == "Citroen"):
+        brand = 'citroen.com'
+        realm = 'clientsB2CCitroen'
+        if (not client_id) or (not client_secret):
+            client_id = "5364defc-80e6-447b-bec6-4af8d1542cae"
+            client_secret = "iE0cD8bB0yJ0dS6rO3nN1hI2wU7uA5xR4gP7lD6vM0oH0nS8dN"
+    elif (manufacturer == "DS"):
+        brand = 'driveds.com'
+        realm = 'clientsB2CDS'
+        if (not client_id) or (not client_secret):
+            client_id = "cbf74ee7-a303-4c3d-aba3-29f5994e2dfa"
+            client_secret = "X6bE6yQ3tH1cG5oA6aW4fS6hK0cR0aK5yN2wE4hP8vL8oW5gU3"
+    elif (manufacturer == "Opel"):
+        brand = 'opel.com'
+        realm = 'clientsB2COpel'
+        if (not client_id) or (not client_secret):
+            client_id = "07364655-93cb-4194-8158-6b035ac2c24c"
+            client_secret = "F2kK7lC5kF5qN7tM0wT8kE3cW1dP0wC5pI6vC0sQ5iP5cN8cJ8"
+    elif (manufacturer == "Vauxhall"):
+        brand = 'vauxhall.co.uk'
+        realm = 'clientsB2CVauxhall'
```

Comment:
Keine der Klammern in dem Beispiel ist nötig. Die könnten alle weg. Noch übersichtlicher fände ich sowas:
```python
ManufacturerConfiguration = NamedTuple(
    "ManufacturerConfiguration",
    [("brand", str), ("realm", str), ("client_id", Optional[str]), ("client_secret", Optional[str])]
)
manufacturer_configurations = {
    "Peugeot": ManufacturerConfiguration(
        brand="peugeot.com",
        realm="clientsB2CPeugeot",
        client_id="1eebc2d5-5df3-459b-a624-20abfcf82530",
        client_secret="T5tP7iS0cO8sC0lA2iE2aR7gK6uE5rF3lJ8pC3nO1pR7tL8vU1",
    ),
    "Citroen": ManufacturerConfiguration(
        brand="citroen.com",
        realm="clientsB2CCitroen",
        client_id="5364defc-80e6-447b-bec6-4af8d1542cae",
        client_secret="iE0cD8bB0yJ0dS6rO3nN1hI2wU7uA5xR4gP7lD6vM0oH0nS8dN",
    ),
    "DS": ManufacturerConfiguration(
        brand="driveds.com",
        realm="clientsB2CDS",
        client_id="cbf74ee7-a303-4c3d-aba3-29f5994e2dfa",
        client_secret="X6bE6yQ3tH1cG5oA6aW4fS6hK0cR0aK5yN2wE4hP8vL8oW5gU3",
    ),
    "Opel": ManufacturerConfiguration(
        brand="opel.com",
        realm="clientsB2COpel",
        client_id="07364655-93cb-4194-8158-6b035ac2c24c",
        client_secret="F2kK7lC5kF5qN7tM0wT8kE3cW1dP0wC5pI6vC0sQ5iP5cN8cJ8",
    ),
    "Vauxhall": ManufacturerConfiguration(
        brand="vauxhall.co.uk",
        realm="clientsB2CVauxhall",
        client_id=None,
        client_secret=None,
    )
}
```
Das kann oben drüber als "Top-Level-Konstante". Dann nurnoch
```python
manufacturer_config = manufacturer_configurations[manufacturer]
```
Und alles ist da...

Reply:
eine letzte if Verzweigung ist geblieben um abzufangen ob in der Konfiguration andere IDs übergeben wurden.